fontir/src/variations.rs

```diff
@@ -0,0 +1,795 @@
+                            / (axis_region.upper - axis_region.peak).into_inner();
+                        axis_region.upper = prev_peak;
+                    }
+                    Ordering::Equal => continue, // can't split in this direction
+                }
+                if ratio > best_ratio {
+                    axis_regions.clear();
+                    best_ratio = ratio;
+                }
+                if ratio == best_ratio {
+                    axis_regions.insert(axis, axis_region);
+                }
+            }
+            for (axis, axis_region) in axis_regions {
+                region.0.insert(axis.to_owned(), axis_region);
+            }
+        }
+        influence.push(region);
+    }
+    influence
+}
+
+/// Figure out the multipliers to use when applying for deltas from masters.
+///
+/// VariationModel::_computeDeltaWeights in Python.
+/// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L438>
+fn delta_weights(
+    locations: &[NormalizedLocation],
+    influencers: &[VariationRegion],
+) -> Vec<Vec<(usize, OrderedFloat<f32>)>> {
+    if log_enabled!(log::Level::Trace) {
+        for (l, i) in locations.iter().zip(influencers) {
+            trace!("{:?}", l);
+            for (axis_name, tent) in i.0.iter() {
+                trace!("  {} {}", axis_name, tent);
+            }
+        }
+    }
+    trace!("Delta Weights");
+    let mut weights = Vec::new();
+    for (loc_idx, location) in locations.iter().enumerate() {
+        weights.push(
+            influencers[..loc_idx]
+                .iter()
+                .enumerate()
+                .filter_map(|(inf_idx, influence)| {
+                    let scalar = influence.scalar_at(location);
+                    if scalar == ZERO {
+                        trace!(
+                            "  no influence: {} {:?} at {:?}",
+                            inf_idx,
+                            influence,
+                            location
+                        );
+                        return None;
+                    }
+                    Some((inf_idx, scalar))
+                })
+                .collect(),
+        );
+        trace!("  {} {:?}", loc_idx, weights.last().unwrap());
+    }
+    weights
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    use ordered_float::OrderedFloat;
+
+    use crate::{
+        coords::{NormalizedCoord, NormalizedLocation},
+        variations::{Tent, ONE},
+    };
+
+    use super::{VariationModel, VariationRegion};
+
+    fn norm_loc(positions: &[(&str, f32)]) -> NormalizedLocation {
+        let mut loc = NormalizedLocation::new();
+        for (axis, value) in positions {
+            loc.set_pos(axis.to_owned(), NormalizedCoord::new(*value));
+        }
+        loc
+    }
+
+    fn tent(lower: f32, peak: f32, upper: f32) -> Tent {
+        Tent {
```

Comment:
oh there's one already, maybe you can change it to take `Into<NormalizedCoord>` params

or maybe not

Reply:
I deliberately didn't implement Into on the coord types because I wanted the conversion to be deliberate, not just an accidental .into that made say a float that was one type into some other.

Reply:
Hm. `From<(f32,f32,f32)>` for Tent might make sense though

Reply:
That feels more idiomatic, good catch. Fixed.